### DISCUSSION TITLE
Screen Masking and Reading Rulers

### DISCUSSION THREAD

**Discussion by AndrewFrazer** [sumatrapdfreader/sumatrapdf#3389] [Ideas]
Hello,

I have started using a reading ruler tool provided by my university and while the software is a pain in the bum and takes over your whole PC, I have found this reading ruler really helps me read papers and reports. I think this feature could really help others with LDs and those who find themselves losing track while reading pdfs.

I love Sumatra PDF and understand that screen masking is already a feature, but I was wondering if this would be considered a valuable addition to the software.

Many thanks!
![ScreenMasking](https://user-images.githubusercontent.com/30623541/227899816-bc683e7d-6bf8-4477-b100-282c2a53cd0d.png)

**Comment by GitHubRulesOK** [comment]
Hmm I got downvoted On overflow for an answer that its Windows Native function both as a screenwide setting
![image](https://user-images.githubusercontent.com/12754913/227924541-ba0bcc74-9fd9-4440-b7fb-e4ae76a0e028.png)

or by using MS Edge any way the discussion went

The academic term is sometimes called RSVP (Rapid Serial Visual Presentation), there are patented hardware and software versions but in principle it is simply a translucent masking added to the viewport. see https://softwarerecs.stackexchange.com/questions/28582/is-there-an-equivalent-to-a-reading-guide-strip-for-windows-os-x-or-linux and http://www.see-n-read.com/products/esee-n-read-2/


![hxN1p](https://user-images.githubusercontent.com/12754913/227919952-37e07bfe-9383-462a-88d0-367f1fa6e299.png)
10 years later and its 2023 so software such as browsers should include such features here is Edge in some sites where Immersive Reader is supported but not StackOverflow !! The above example is using an edge extension. https://microsoftedge.microsoft.com/addons/detail/screen-mask/dfanfcmhbdocjfpmnoebccndgmhlincl others are available for other browsers https://chrome.google.com/webstore/detail/reading-ruler/phiedfcbjfjagnjikfbobmldbpmdcpfk To get the Reader Mode options on Chrome: or Edge look at the available flags

Some PDF readers like Mac Skim include such accessibility option.

However, simplest is :-

Most PDF readers can be reduced to focus viewport on single lines and with auto scrolling that allows for more focused "line by line" reading without the audio, plus fast and easy adjustments/enlarging for PDF variable lines with illustrations.

![1L8dS](https://user-images.githubusercontent.com/12754913/227920793-3b4c9deb-5a7a-4e28-be20-b9eb03f910aa.png)

Note as per above PDF where much of the text is actually one or two lines out of order it is not trivial for a PDF reader to understand which text base line is independently to be used next. in reality "Read Aloud" will read two variable height lines then jump to top of page then back to the second visible line. PDF lines are not the visible order nor a constant height/spacing, you might expect.

**Comment by GitHubRulesOK** [comment]
Well I bothered to use this request as a test case for following the canvas in native Windows and came up with a proof of concept, 
It is only tested on Windows 7 & 10 but likely to work on 11 ?
To see it in action you need to **download and run**  https://github.com/GitHubRulesOK/SumatraPDF-Plus/blob/master/FuBar.cmd which generates FuBar.exe and that can be run using SumatraPDF ExternalViewers or a menu shortcut.

But if you wish to edit the settings in MS Notepad you need to **rerun the cmd** file without the exe active then restart.

<img width="1239" height="455" alt="image" src="https://github.com/user-attachments/assets/87eb96c0-129b-436a-8c8b-eddf238ee4ec" />

**Comment by GitHubRulesOK** [comment]
**[Updated]** https://github.com/GitHubRulesOK/SumatraPDF-Plus/blob/master/FuBar.cmd
Ok after download just edit mask colours to preferred as the aperture width and fog can be adjusted by sliders after you DoubleClick the cmd file and add as plug-in command. Note the aperture near top and bottom of canvas now slide to the edges.
<img width="1092" height="746" alt="image" src="https://github.com/user-attachments/assets/f0280e06-dce5-4b74-a1a6-a2c9cc5e7401" />

<img width="959" height="486" alt="image" src="https://github.com/user-attachments/assets/ed3b8feb-8f8c-4542-b34d-1b063da63c96" />

**Comment by kjk** [comment]
Implemented as **Reading Bar** (View menu / `CmdToggleReadingBar`). Invert mode is the screen mask from this request; the default is a colored highlight band. Drag to move, edges to resize. Settings are in `ReadingBar`. See #5771.

https://github.com/sumatrapdfreader/sumatrapdf/commit/c21bfcaf1
